builtin/providers/aws/resource_aws_vpc_dhcp_options.go

```diff
@@ -121,7 +121,7 @@ func resourceAwsVpcDhcpOptionsCreate(d *schema.ResourceData, meta interface{}) e
 	log.Printf("[DEBUG] Waiting for DHCP Options (%s) to become available", d.Id())
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{"pending"},
-		Target:  "",
+		Target:  []string{},
```

Comment:
final follow up; I opened https://github.com/hashicorp/terraform/pull/4828 as a "fix". The bug only presents itself if `[]string{}` is used in _creation_. It's normal for `nil` to be returned when we're deleting something, but DHCP Options here doesn't really provide a status.